### PR TITLE
Full aggregate support: review suggestions

### PR DESCRIPTION
**Pull request by AntoniusW**
This PR bundles several suggested code changes on PR #268 in the Alpha repository.

### CODE REVIEW

**Comment by AntoniusW** [comment]
@madmike200590  please have a look at this PR again. It should include all changes as we discussed. Also contains some unit tests and a disabled one triggering an issue with complex aggregate dependencies via global variables.

File: src/main/java/at/ac/tuwien/kr/alpha/grounder/transformation/NormalizeProgramTransformation.java [hunk 1]
```diff
@@ -34,8 +36,6 @@ public NormalProgram apply(InputProgram inputProgram) {
 		NormalProgram retVal = NormalProgram.fromInputProgram(tmpPrg);
 		// Transform intervals - CAUTION - this MUST come before VariableEqualityRemoval!
```
Comment (madmike200590) [inline src/main/java/at/ac/tuwien/kr/alpha/grounder/transformation/NormalizeProgramTransformation.java]: I think this comment is no longer true and should be removed.

**Review by madmike200590 — APPROVED** [review]
Thanks, apart from one (very nit-picky) comment that doesn't justify a re-review after resolution, this looks great!

File: src/main/java/at/ac/tuwien/kr/alpha/grounder/transformation/NormalizeProgramTransformation.java (hunk 1 above)
Comment (AntoniusW) [reply src/main/java/at/ac/tuwien/kr/alpha/grounder/transformation/NormalizeProgramTransformation.java]: Fixed.

**Pull request merged**